app/src/main/kotlin/compiler/ast/AstNodeTypes.kt

```diff
@@ -0,0 +1,6 @@
+package compiler.ast
+
+object AstNodeTypes {
```

Comment:
Is this `object` necessary? The interfaces below can be top-level (and we can assume a convention that their names end with `Node`).